DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -268,6 +268,8 @@ bool pat::PackedCandidate::longLived() const {return false;}
 
 bool pat::PackedCandidate::massConstraint() const {return false;}
 
+// puppiweight
+void pat::PackedCandidate::setPuppiWeight(float p) { packedPuppiweight_ = pack8log(p,-2,0);}
```

Comment:
how about making it symmetric in -1,1 instead? so that it has higher precision near zero too?
pack8log( (p-0.5)*2,-2,0,64)
where the last argument (64) is needed if you want to take the same space as it is doing now with 7+1 signed bits (i.e. one is never used)